frontend/src/app/api/callback/route.ts

```diff
@@ -16,16 +16,18 @@ export const GET = async (request: NextRequest) => {
     });
   }
 
-  const secret = cookies().get("oauth_token_secret")?.value;
```

Comment:
Dobrze rozumiem, że wcześniej funkcja cookies() za każdym razem była wywoływana, a teraz tylko raz i później z tego jednego wywołania odczytujesz?

Reply:
Po zmianie na nexta 15 wprowadzil asynchroniczne funkcje takie jak headers i cookies ktore wczesniej byly statyczne, dlatego aktualnie tworze jedna zmienna ktora awaituje to
`const cookies = await cookiesPromise();` 
i poziej uzywam tej zmiennej `cookies`. Rownie dobrze mozna byc robic cos w stylu:
`(await cookies()).get("oauth_token_secret")?.value`
ale tak jest chyba bardziej czytelniej

Reply:
Oki, teraz kumam, to faktycznie twoje rozwiązanie bardziej czytelne